include/zcbor_decode.h

```diff
@@ -146,6 +150,92 @@ bool zcbor_uint64_expect_union(zcbor_state_t *state, uint64_t expected);
  */
 bool zcbor_list_start_decode(zcbor_state_t *state);
 bool zcbor_map_start_decode(zcbor_state_t *state);
+bool zcbor_unordered_map_start_decode(zcbor_state_t *state);
+
+/** Search for a key in a map.
+ *
+ * By spec, elements (key-value pairs) in CBOR maps can appear in any order.
+ * Use this function when you don't know the order of the elements of a map.
+ *
+ * This must only be used while inside a map that has been entered via
+ * @ref zcbor_unordered_map_start_decode. Use @ref zcbor_unordered_map_end_decode
+ * when leaving the map.
+ *
+ * When this function has returned, it can never immediately be called again
+ * because the @p state is pointing at a value, but this function needs it to
+ * point to a key. Decode the value, or skip it (@ref zcbor_any_skip) before
+ * calling this function again.
+ *
+ * Whether and how elements found with this function are marked as found,
+ * depends on the state of @ref manually_process_elem and ZCBOR_MAP_SEARCH_FLAGS.
+ *
+ * @note When ZCBOR_MAP_SEARCH_FLAGS is not defined,
```

Comment:
If the only purpose of this flag is to save the state of whether elements are processed, then I think we should rename it. `SEARCH_FLAGS` makes it sound like you set some options for how the processing and searching is performed.

Reply:
The search_flags only consist of one flag per element, there are for example no "option" flags for controlling processing and searching. The search flags are checked by zcbor_unordered_map_search() when searching, and zcbor_unordered_map_end_decode() when closing a map. Do you have a name suggestion? I feel blind to the name having gotten used to it.